### PR TITLE
K8SPXC-454: add certificate checks

### PR DESCRIPTION
**Pull request by qJkee**
[K8SPXC-454](https://jira.percona.com/browse/K8SPXC-454)

Check if certificates exist before deploying pxc cluster

### CODE REVIEW

File: pkg/controller/pxc/controller.go [hunk 1]
```diff
@@ -562,8 +562,8 @@ func (r *ReconcilePerconaXtraDBCluster) deploy(cr *api.PerconaXtraDBCluster) err
 	}
 
 	sslInternalHash, err := r.getSecretHash(cr, cr.Spec.PXC.SSLInternalSecretName, cr.Spec.AllowUnsafeConfig)
-	if err != nil && !k8serrors.IsNotFound(err) {
-		return fmt.Errorf("get secret hash error: %v", err)
+	if err != nil {
```
Comment (delgod) [inline pkg/controller/pxc/controller.go:565]:
```suggestion
	if err != nil && k8serrors.IsNotFound(err) && allowNonExistingSecret {
```

**Review by delgod — CHANGES_REQUESTED** [review]

File: pkg/controller/pxc/controller.go (hunk 1 above)
Comment (qJkee) [reply pkg/controller/pxc/controller.go:565]: we have exactly the same check inside getSecretHash func

Comment (delgod) [reply pkg/controller/pxc/controller.go:565]: got it, thanks

**Review by delgod — APPROVED** [review]

File: pkg/controller/pxc/upgrade.go [hunk 2]
```diff
@@ -477,6 +477,11 @@ func (r *ReconcilePerconaXtraDBCluster) getSecretHash(cr *api.PerconaXtraDBClust
 		return "", err
 	}
 
+	if secretName == cr.Spec.PXC.SSLSecretName || secretName == cr.Spec.PXC.SSLInternalSecretName{
+		if secretObj.Data["ca.crt"] == nil || secretObj.Data["tls.crt"] == nil || secretObj.Data["tls.key"] == nil {
+			return "", errors.New("one or more tls certificates are not found in secret")
+		}
+	}
```
Comment (blez) [inline pkg/controller/pxc/upgrade.go:484]: I would move this code to separate func. The reason is that this code break single responsibility principle. I mean function was written to get hashes but now it tries to validate particular certificates.

**Review by blez — APPROVED** [review]

File: pkg/controller/pxc/upgrade.go (hunk 2 above)
Comment (qJkee) [reply pkg/controller/pxc/upgrade.go:484]: kk, i'll rewrite this part with separate pr

**Review by hors — APPROVED** [review]

**Pull request merged**